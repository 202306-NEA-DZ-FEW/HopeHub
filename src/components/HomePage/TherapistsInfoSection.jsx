import Image from "next/image";
import { useTranslation } from "next-i18next";
import React from "react";

import { useAppcontext } from "@/context/state";

import BookingButton from "../BookingButton/BookingButton";
import therapists from "../../../public/assets/therapists.svg";

export default function TherapistsInfoSection({ user }) {
    //Function used for translations
    const { t } = useTranslation("common");

    //Displaying the therapists info section
    return (
        <div className='w-full py-4 bg-Primary dark:bg-Dark_Primary lg:pb-4'>
            <h1 className='mx-6 mb-2 text-base md:mb-4 md:text-3xl md:mx-9 md:mt-10 font-poppins uppercase font-medium inline-block text-NeutralBlack dark:text-NeutralWhite '>
                {t(
                    "Professional, licensed, and vetted therapists that you can trust."
                )}
            </h1>

            <div className='mx-6 w-64 md:w-96 lg:w-auto md:mx-9'>
                <Image
                    src={therapists}
                    alt='therapists'
                    width={500}
                    height={60}
                    layout='fixed'
                />
            </div>
            <p className='text-xs mx-6 my-4 md:text-xl md:mx-9 font-normal text-NeutralBlack dark:text-NeutralWhite font-poppins'>
                {t(
                    "Tap to the world's largest network of licensed, accredited, and experienced therapists who can help you with a range of issues including depression, anxiety, relationships, trauma, grief, and more. with our therapists, you get the same professionalism and quality you would expect from an in-office therapist, but with the ability to communicate whenever and however you want."
                )}
            </p>

            {/* Adding the button for booking an appointment */}
            <div className='inset-0 flex items-end justify-end my-2 md:my-8 mx-6 md:mx-10 mb-6 sm:mx-20 lg:mx-10'>
                <BookingButton
                    destination={
                        user !== undefined && user.isTherapist
                            ? `/call?userid=${user.uid}` // If user is a therapist, go to call page
                            : user !== undefined
                            ? `/booking?userid=${user.uid}` // If user is a patient, go to booking page
                            : "/Auth" // If user is not identified, redirect to auth
                    }
                    buttonText={
                        user !== undefined && user.isTherapist
                            ? "Join Call" // If user is a therapist, show 'Join Call'
                            : user !== undefined
                            ? "Book An Appointment" // If user is a patient, show 'Book An Appointment'
                            : "Book An Appointment"
                    }
                />
            </div>
        </div>
    );
}
