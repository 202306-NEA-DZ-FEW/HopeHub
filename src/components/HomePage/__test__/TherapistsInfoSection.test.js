import renderer from "react-test-renderer";

import TherapistsInfoSection from "../TherapistsInfoSection";

it("renders correctly", () => {
    const tree = renderer.create(<TherapistsInfoSection />).toJSON();
    expect(tree).toMatchSnapshot();
});
