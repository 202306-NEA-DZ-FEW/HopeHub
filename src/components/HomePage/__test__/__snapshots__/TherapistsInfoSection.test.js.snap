// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-full py-4 bg-Primary dark:bg-Dark_Primary lg:pb-4"
>
  <h1
    className="mx-6 mb-2 text-base md:mb-4 md:text-3xl md:mx-9 md:mt-10 font-poppins uppercase font-medium inline-block text-NeutralBlack dark:text-NeutralWhite "
  >
    Professional, licensed, and vetted therapists that you can trust.
  </h1>
  <div
    className="mx-6 w-64 md:w-96 lg:w-auto md:mx-9"
  >
    <img
      alt="therapists"
      data-nimg="1"
      decoding="async"
      height={60}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={500}
    />
  </div>
  <p
    className="text-xs mx-6 my-4 md:text-xl md:mx-9 font-normal text-NeutralBlack dark:text-NeutralWhite font-poppins"
  >
    Tap to the world's largest network of licensed, accredited, and experienced therapists who can help you with a range of issues including depression, anxiety, relationships, trauma, grief, and more. with our therapists, you get the same professionalism and quality you would expect from an in-office therapist, but with the ability to communicate whenever and however you want.
  </p>
  <div
    className="inset-0 flex items-end justify-end my-2 md:my-8 mx-6 md:mx-10 mb-6 sm:mx-20 lg:mx-10"
  >
    <div>
      <button
        className="w-28 md:w-72 h-10 rounded-md text-xs sm:text-base lg:text-xl font-poppins font-regular bg-Accent text-NeutralBlack dark:text-NeutralWhite uppercase   dark:bg-Dark_Accent dark:hover:bg-[#3E4E68]  hover:bg-[#879AB8] hover:text-NeutralWhite hover:scale-105 duration-500"
        data-testid="booking-button"
      >
        <a
          href="/Auth"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Book An Appointment
        </a>
      </button>
    </div>
  </div>
</div>
`;
